docs/proposals/20200720-single-controller-multitenancy.md

```diff
@@ -0,0 +1,548 @@
+The ClusterScope contains all of the infromation needed to make an authenticated request against an
+Azure cloud. The `Authorizer` currently contains the AAD identity information loaded from the CAPZ
+controller environment. The `Authorizer` is used to fetch and refresh tokes for all Azure clients.
+
+```go
+// AzureClients contains all the Azure clients used by the scopes.
+type AzureClients struct {
+	SubscriptionID             string
+	ResourceManagerEndpoint    string
+	ResourceManagerVMDNSSuffix string
+	Authorizer                 autorest.Authorizer
+}
+
+```
+
+The signatures for the functions which create these instances are as follows:
+
+```go
+func NewClusterScope(params ClusterScopeParams) (*ClusterScope, error) {
+  ...
+  return &ClusterScope{
+    ...
+  }, nil
+}
+```
+
+#### Proposed Changes
+
+##### Cluster API Provider Azure v1alpha3 types
+
+<strong><em>Changed Resources</strong></em>
+* `AzureCluster`
+
+<strong><em>New Resources</strong></em>
+
+<em>Cluster scoped resources</em>
+
+* `AzureServicePrincipal` represents an AAD Service Principal. Should support both 
+   certificate and text secrets.
+* `AzureSystemAssignedIdentity` represents an Azure System Assigned Identity provided via 
+   Pod Identity
+* `AzureUserAssignedIdentity` represents an Azure User Assigned Identity provided via Pod 
+   Identity
+
+<strong><em>Changes to AzureCluster</em></strong>
+
+A new field is added to the `AzureClusterSpec` to reference a principal. We intend to use 
+`corev1.LocalObjectReference` in order to ensure that the only objects that can be references are 
+either in the same namespace or are scoped to the entire cluster.
+
+```go
+// AzurePrincipalKind defines allowed Azure principal types
+// +kubebuilder:validation:Enum=AzureSystemAssigned;AzureUserAssigned;AzureServicePrincipal
+type AzurePrincipalKind string
+
+type AzurePrincipalRef struct {
+  Kind AzurePrincipalKind `json:"kind"`
+  Name string `json:"name"`
+}
+
+type  AzureClusterSpec  struct {
+  ...
+  // +optional
+  PrincipalRef *AzurePrincipalRef `json:"principalRef,omitempty"`
+```
+
+Example:
+
+```yaml
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AzureCluster
+metadata:
+  name: "test"
+  namespace: "test"
+spec:
+  region: "westus2"
+  principalRef:
+    kind: AzureServicePrincipal
+    name: test-account
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AzureServicePrincipal
+metadata:
+  name: "test-account"
+spec:
+  secretRef:
+    name: test-account-creds
+    namespace: capz-system
+  allowedNamespaces:
+  - "test"
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: "test-account-creds"
+  namespace: capz-system
+stringData:
+ tenantID: 6bec3eaa-cac6-11ea-87d0-0242ac130003
+ clientID: 74e870e6-cac6-11ea-87d0-0242ac130003
+ clietSecret: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+```
+
+<em>AzureSystemAssignedIdentity</em>
+
+`AzureSystemAssignedIdentity` allows CAPZ to use the SystemAssigned identity provided via Pod 
```

Comment:
Is this for using Managed identity for CAPZ provisioning? I assume that's seperate from VM identity to be used by the cloud provider? (just making sure my understanding is correct)

Reply:
Yes. I added an explanation and scoping of the usage of AAD Pod Identity.